src/main/java/com/atlauncher/workers/InstanceInstaller.java

```diff
@@ -2470,6 +2470,7 @@ private void saveInstanceJson() {
 
         if (!this.isReinstall) {
             instanceLauncher = new InstanceLauncher();
+            InstanceManager.removeInstance(this.instance); // Remove old instance from app data, as it no longer exists
```

Comment:
Is this in the right place? The condition is `!this.isReinstall` so it's run only on a brand new instance being installed

Reply:
Honestly I have no clue right now, I might be too sleep deprived.

Let me look at this problem again